### PR TITLE
Dashboard Charts v1.0.0

### PR DESCRIPTION
**Pull request by samussiah**

### CODE REVIEW

**Comment by samussiah** [comment]
@pburnsdata , this might look like a lot, and it is, but not _138-files-a-lot_.  Focus on the stuff in onInit in the accrual charts and the new features in onResize.

File: src/enrollment/callbacks/onResize/addBarClick.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                        'font-weight': 'bold',
+                    })
+                    .text(`Displaying ${d.values.x} ${d.key} participants at ${d.values.y}`);
+                this.table.backButton = this.table.container
+                    .append('button')
+                    .style({
+                        float: 'right',
+                    })
+                    .text(`Back`)
+                    .on('click', () => {
```
Comment (pburnsdata) [inline src/enrollment/callbacks/onResize/addBarClick.js]: this is really nicely done

File: src/enrollment/callbacks/onResize/addBarClick.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
+export default function addBarClick() {
+    const context = this;
+
+    this.marks.forEach(mark => {
+        this.svg
+            .selectAll(`.wc-data-mark.${mark.type}`)
+            .style({
+                cursor: 'pointer'
+            })
+            .on('click', d => {
```
Comment (pburnsdata) [inline src/enrollment/callbacks/onResize/addBarClick.js]: this is really nicely done

File: src/enrollment/callbacks/onInit/useSiteAbbreviation.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import defineSet from '../../../util/defineSet';
+
+export default function useSiteAbbreviation() {
+    this.config.useSiteAbbreviation = this.raw_data[0].hasOwnProperty(this.config.site_abbreviation_col);
+    if (this.config.useSiteAbbreviation) {
```
Comment (pburnsdata) [inline src/enrollment/callbacks/onInit/useSiteAbbreviation.js]: smart

File: src/util/defineSet.js [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
+    const set = d3.set(this.raw_data.map(d => variables.map(variable => `${variable}-:-${d[variable]}`).join(':|:')))
         .values()
         .map(value => {
             return value
                 .split(':|:')
                 .reduce(
                     (acc,cur) => {
-                        const key = cur.split(':')[0];
-                        const value = cur.split(':')[1];
+                        const key = cur.split('-:-')[0];
```
Comment (pburnsdata) [inline src/util/defineSet.js]: safe

File: src/accrual-over-time/callbacks/onResize/customizeTargetLines.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+export default function customizeTargetLines() {
+    // Update lines in chart.
+    this.marks
+        .filter(mark => mark.type === 'line')
+        .forEach(mark => {
+            mark.paths.each(function(d) {
+                if (/target/i.test(d.key))
+                    this.setAttribute('stroke-dasharray', '2 4');
```
Comment (pburnsdata) [inline src/accrual-over-time/callbacks/onResize/customizeTargetLines.js]: love it

File: src/accrual-over-time-derived/callbacks/onInit/deriveData.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
+export default function deriveData() {
```
Comment (pburnsdata) [inline src/accrual-over-time-derived/callbacks/onInit/deriveData.js]: wow  - total bonus chart!

File: src/accrual/data-guidelines-wiki.md [hunk 7]
```diff
@@ -1,7 +1,7 @@
 The Accrual chart accepts [JSON](https://en.wikipedia.org/wiki/JSON) data of the format returned by [`d3.csv()`](https://github.com/d3/d3-3.x-api-reference/blob/master/CSV.md). It plots the number of participants in each study populations by site.
 
 ## Data structure
-one record per participant per population
+one record per participant per population with a discrete variable that will plot on the y-axis
```
Comment (pburnsdata) [inline src/accrual/data-guidelines-wiki.md]: yep - good addition

**Comment by emmorris** [comment]
No regression testing needed.  Feature testing was essentially regression testing for this case because we tested everything merged together for this first release of the two charts.

**Pull request merged**